### DISCUSSION TITLE
How to reliably check if top-level field

### DISCUSSION THREAD

**Discussion by Wedrix** [webonyx/graphql-php#1224] [Q&A] [answered]
Hello. Any ideas on how I can reliably check if a field is a top-level field in a resolver? Preferably using the `$info` parameter. I considered matching the field name to either 'mutation,' 'query,' or 'subscription,' but that's not guaranteed since it's possible for the schema to have fields with these names. Another alternative is to compare the `$root` param with the `$rootValue` passed in `GraphQL::executeQuery()` but that is also not guaranteed. Was wondering if there is a property I can check on the parent type `$info->parentType` or something along those lines?

**Comment by spawnia** [comment]
https://github.com/webonyx/graphql-php/blob/e411398ef4b76542c4670fd1560a928de5d08cbc/src/Type/Definition/ResolveInfo.php#L57-L62

Use `$resolveInfo->parentType->name` and check if it is equal to `Query`, `Mutation` or `Subscription` - works as long as you do not change the default names of the root types.

**Reply by Wedrix** [reply]
Thanks for the reply, @spawnia, but what if the schema has Object types with those names? It seems the [spec](https://spec.graphql.org/October2021/#sec-Names.Reserved-Names) does not reserve them. I'm working on a package therefore I cannot guarantee such types will not be defined in the schema.

**Reply by spawnia** [reply]
I guess you would have to get a reference to the users `Schema` and compare against the root types configured there. Alternatively, add some custom configuration option or simply do not support non-default root types.

**Reply by Wedrix** [reply]
@spawnia, what do you think of this approach `count($this->info->path) === 1`? Seems like it will only work for top-level fields.

**Reply by spawnia** [reply]
That seems entirely reasonable! I would be open to accepting a pull request to add a method or field to `ResolveInfo` to make this easier.

**Comment by Wedrix** [comment, marked as answer]
count($this->info->path) === 1
